PowershellTools.HostService/Program.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.Net.Security;
+using System.ServiceModel;
+using System.Threading;
+using PowerShellTools.Common;
+using PowerShellTools.Common.ServiceManagement.IntelliSenseContract;
+using PowerShellTools.HostService.ServiceManagement;
+
+namespace PowerShellTools.HostService
+{
+    internal class Program
+    {
+        private static ServiceHost _powershellServiceHost;
+        private static AutoResetEvent _processExitEvent;
+
+        [LoaderOptimization(LoaderOptimization.SingleDomain)]
+        internal static int Main(string[] args)
+        {
+            if (args.Length != 3 ||
+                !(args[0].StartsWith(Constants.UniqueEndpointArg, StringComparison.OrdinalIgnoreCase)
+                && args[1].StartsWith(Constants.VsProcessIdArg, StringComparison.OrdinalIgnoreCase)
+                && args[2].StartsWith(Constants.ReadyEventUniqueNameArg, StringComparison.OrdinalIgnoreCase)
+                ))
+            {
+                return 1;
+            }
+
+            _processExitEvent = new AutoResetEvent(false);
+
+            string endpointGuid = args[0].Remove(0, Constants.UniqueEndpointArg.Length);

```

Comment:
args[0] is a concatenation of Constants.UniqueEndpointArg and a Guid. Besides, I'm not trying o get a Guid.